File: 백준/Silver/1946. 신입 사원/신입 사원.py
import sys

input = sys.stdin

T = int(input.readline())
for _ in range(T):
    n = int(input.readline())
    new_employees = []
    answer = 1
    for i in range(n):
        new_employees.append(list(map(int, input.readline().split())))
    new_employees.sort()
    temp = new_employees[0][1]
    for i in range(1, n):
        if temp > new_employees[i][1]:
            answer += 1
            temp = new_employees[i][1]
    print(answer)

File: 백준/Silver/1946. 신입 사원/신입 사원.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.ArrayList;
import java.util.Collection;
import java.util.Collections;
import java.util.List;
import java.util.StringTokenizer;

public class Main {
    public static void main(String[] args) throws IOException{
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        int t = Integer.parseInt(br.readLine());

        StringBuilder answer = new StringBuilder();

        while(t-- > 0){
            int n = Integer.parseInt(br.readLine());
            List<int[]> employees = new ArrayList<>();

            for(int i=0 ; i<n ; i++){
                StringTokenizer stk = new StringTokenizer(br.readLine());
                int document = Integer.parseInt(stk.nextToken()), interview = Integer.parseInt(stk.nextToken());

                employees.add(new int[]{document, interview});
            }

            Collections.sort(employees, (o1, o2) -> o1[0] - o2[0]);

            int prev = employees.get(0)[1];
            int count = 0;

            for(int i=0 ; i <n ; i++){
                if(prev >= employees.get(i)[1])
                    count++;

                prev = Math.min(prev, employees.get(i)[1]);
            }

            answer.append(count).append("\n");
        }

        System.out.println(answer);
    }    
}
